test/jdk/java/awt/print/PrinterJob/PrintImage.java

```diff
@@ -21,276 +21,167 @@
-
-                if(printerJob.printDialog()){
-                        try {
-                                printerJob.print();
-                        }
-                        catch(PrinterException p){
-                        }
-                }
-                else
-                        printerJob.cancel();
-        }
+        printerJob.setPrintable(printImageCanvas, pageFormat);
 
+        if (printerJob.printDialog()) {
+            try {
+                printerJob.print();
+            } catch (PrinterException e) {
+                e.printStackTrace();
+            }
+        } else
+            printerJob.cancel();
+    }
 }
 
 class PrintImageCanvas extends Canvas implements Printable {
 
-        private PrintImage pdsFrame;
-
-        public PrintImageCanvas(PrintImage pds) {
-                pdsFrame = pds;
-        }
-
-        public void paint(Graphics g) {
-                Font drawFont = new Font("MS Mincho",Font.ITALIC,50);
-                g.setFont(drawFont);
-                g.drawString("PrintSample!",100,150);
-        }
-
-        public int print(Graphics g, PageFormat pf, int pi)
-                throws PrinterException {
-
-                if(pi>=1)
-                        return NO_SUCH_PAGE;
-                else{
-                        Graphics2D g2 = (Graphics2D)g;
-                        g.setColor(new Color(0,0,0,200));
-
-                        Font drawFont = new Font("MS Mincho",Font.ITALIC,50);
-                        g.setFont(drawFont);
-                        g.drawString("PrintSample!",100,150);
-                        return PAGE_EXISTS;
-                }
-        }
-}
-
-
-class Sysout {
-   private static TestDialog dialog;
-
-   public static void createDialogWithInstructions( String[] instructions )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      dialog.printInstructions( instructions );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
+    public PrintImageCanvas(PrintImage pds) {
     }
 
-
-   public static void printInstructions( String[] instructions )
-    {
-      dialog.printInstructions( instructions );
+    public void paint(Graphics g) {
+        Font drawFont = new Font("MS Mincho", Font.ITALIC, 50);
+        g.setFont(drawFont);
+        g.drawString("PrintSample!", 100, 150);
     }
 
+    public int print(Graphics g, PageFormat pf, int pi)
+            throws PrinterException {
 
-   public static void println( String messageIn )
-    {
-      dialog.displayMessage( messageIn );
-    }
+        if (pi >= 1)
+            return NO_SUCH_PAGE;
+        else {
```

Comment:
Use braces.

The `else` block is redundant: you return from the body of `if`. Remove the `else` to decrease the level of indentation.